### ISSUE TITLE
list index out of range

### ISSUE THREAD

**Issue by jintianyun56** [Anan-Wu-XMU/SVM-M#3]
Hi Prof. Anan,

It showed "list index out of range" when I run the evanmr2.py .
also show "Explicit valence for atom # 23 N, 4, is greater than permitted"

Would you like to help me to find something to troubleshoot?

Thanks

**Comment by Anan-Wu-XMU** [comment]
Could you please check whether there are data in allNMR and allHNMR, in which calculated 13C/1H chemical shieldings are collected. evanmr2.py read relevant data from allNMR/allHNMR, freeG.txt and automatically evaluate the Boltzmann averaged 13C/1H chemical shifts, which will be stored in avNMR.txt and avHNMR.txt, respectively. It looks to me that there are problems in NMR calculations or possibly geometry optimizations.

**Comment by jintianyun56** [comment]
Thanks,
Yes,  avNMR.txt and avHNMR.txt were blank.
I'll try to make it work.


Anan-Wu-XMU ***@***.***> 于2023年12月10日周日 17:08写道：

**Comment by Anan-Wu-XMU** [comment]
A success run will generate files as shown in the following figure, in which

      *.sdf :   input conformers 
      *geom?.com/log:   geometry optimization files
      *geom?.new:   optimized geometries
      *geom-NMR.com/log:   xOPBE NMR calculation files
      *geom?-NMR.NMR:  calculated 13C chemical shieldings of individual conformer
      *geom?-NMR.HNMR: calculated 1H chemical shieldings of individual conformer 
      allNMR/allHNMR:   collected chemical shieldings for all valid conformers (< 10kj/mol)
      avNMR.txt:    the Boltzmann averaged 13C chemical shifts
      avHNMR.txt:   the Boltzmann averaged 1H chemical shifts 

![2023-12-12 091306](https://github.com/Anan-Wu-XMU/SVM-M/assets/89952471/1a36232d-4343-4649-9055-c650ea99bd70)


> Thanks, Yes, avNMR.txt and avHNMR.txt were blank. I'll try to make it work.

**Comment by jintianyun56** [comment]
Thanks, The example ran well, and I am trying my structure.
I am wondering about why we don't assign the chemical shit to the carbon
number. the svm method only sequences the chemical shift from small to big,
as experiments C2: 132, C3 134, if case 1 calculation C2:134, C3: 132,;
case 2 cal. C2:132, C3: 134. I think the SVM-M method is hard to identify
case 1 and 2.

Anan-Wu-XMU ***@***.***> 于2023年12月11日周一 17:28写道：

**Comment by Anan-Wu-XMU** [comment]
Indeed, case 1 and 2 are identical within the framework of SVM-M method. As stated in our paper, the sorting is to avoid the mis-assignment in experiments and make the model applicable when only unassigned data are available.  We are currently working on a 2D method based on SVM-M, which can handle 2D NMR data. The atomic labelling are important and taken in consideration in this method.  

> Thanks, The example ran well, and I am trying my structure. I am wondering about why we don't assign the chemical shit to the carbon number. the svm method only sequences the chemical shift from small to big, as experiments C2: 132, C3 134, if case 1 calculation C2:134, C3: 132,; case 2 cal. C2:132, C3: 134. I think the SVM-M method is hard to identify case 1 and 2.

**Comment by jintianyun56** [comment]
Thanks, I am looking forward to the new 2D method.

Anan-Wu-XMU ***@***.***> 于2023年12月11日周一 23:31写道：
